### PR TITLE
Add support for OpenSSL 3 (plus some general recipe maintenence)

### PR DESCRIPTION
**Pull request by dean0x7d**
Python versions >= 3.10 support OpenSSL 3. This PR makes `openssl` v3.1.1 the default for those versions (Python 3.9 gets v1.1.1u by default). Some extra care was needed for static linking. See the code comment for details/tradeoffs.

This PR also does a bit of maintenance to remove deprecation warnings. It also adds the `EmbeddedPython_EXECUTABLE` variable to CMake as we found this would be helpful in one project. The code comment and changelog contain a lot more detail, but in short `Python_EXECUTABLE` points to the `core` Python with only the standard library (the variable name is not up to us as it needs to be what CMake's builtin `find_package(Python)` provides). `EmbeddedPython_EXECUTABLE` points to the full Python env with `pip` packages (the naming is up to us, feel free to provide suggestions).

### CODE REVIEW

File: core/conanfile.py [hunk 1]
```diff
@@ -64,14 +62,10 @@ def requirements(self):
...
                 self.requires("mpdecimal/2.5.0")
 
-        # The pre-conan-center-index version of `openssl` was capitalized as `OpenSSL`.
-        # Both versions can't live in the same Conan cache so we need this compatibility
-        # option to pick the available version. The cache case-sensitivity issue should
-        # be solved in Conan 2.0, but we need this for now.
-        if self.options.openssl_variant == "lowercase":
-            self.requires("openssl/1.1.1k")
+        if self.pyversion >= scm.Version("3.10.0"):
+            self.requires("openssl/3.1.1")
```
Comment (agom) [inline core/conanfile.py:66]: `openssl/3.1.1` [crashes](https://github.com/openssl/openssl/issues/20753) on MacOS arm when running with a debugger attached, the fix has been [merged](https://github.com/openssl/openssl/pull/21261) but it's not released yet. I'd recommend either going for `3.0.8` or waiting for `3.1.2`

**Review by agom — CHANGES_REQUESTED** [review]
Looks good, the only thing is the `openssl` 3 version, see the comment for the details.

File: core/conanfile.py (hunk 1 above)
Comment (dean0x7d) [reply core/conanfile.py]: Great catch! I've change the version and added a comment to reflect this.

**Review by agom — APPROVED** [review]
Great stuff, I'd be in favor of tagging a 2.0 an drop the `lowercase` flag altogether but can be done at a later stage.

**Pull request merged**